common/lib/xmodule/xmodule/html_module.py

```diff
@@ -440,8 +440,7 @@ def get_html(self):
             return self.data
         else:
             course_updates = [item for item in self.items if item.get('status') == self.STATUS_VISIBLE]
-            course_updates.sort(key=lambda item: CourseInfoModule.safe_parse_date(item['date']), reverse=True)
-
+            course_updates.sort(key=lambda item: (datetime.strptime(item['date'], '%B %d, %Y'), item['id']), reverse=True)
```

Comment:
It would be safer to do `CourseInfoModule.safe_parse_date(item['date'])` instead of `datetime.strptime(item['date'], '%B %d, %Y')` directly